docs/csharp/how-to/modify-string-contents.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "How to: Modify String Contents (C# Guide)"
+ms.date: 02/26/2018
+ms.prod: .net
+ms.technology: 
+  - "devlang-csharp"
+ms.topic: "article"
+helpviewer_keywords: 
+  - "strings [C#], modifying"
+author: "BillWagner"
+ms.author: "wiwagn"
+---
+
+# How to: Modify String Contents (C# Programming Guide)
+
+You can't modify the contents of a string after it has been created. The <xref:System.String?displayProperty=nameWithType> class is *immutable*. The examples in this article show how to create a new string object that represents the result of some change to an existing string.
```

Comment:
I did make a few updates here based on your comment.

First, I added "typically" to the first sentence.  I tried introducing the term "unsafe code" there, but I rejected it because the article is not really about using unsafe code, but about how to change the value of strings.

I want to continue to use the term "Modifying strings" because that is a more common search term than "transforming". I agree it is a bit inaccurate, and the content of the article works to make that clear.

Next, I added a bit more explanatory text in the paragraph that introduces the unsafe sample.

I hope that adds the clarity we need.